packages/web3-utils/src/index.js

```diff
@@ -328,36 +328,52 @@ var toChecksumAddress = function (address) {
  * @returns {Number} -1, 0, or 1
  */
 var compareBlockNumbers = function(a, b) {
+    // Increasing order: (genesis = earliest), safe, (finalized ~ latest), pending
+    // safe vs block-num cant be compared as block number provided can be on left or right side of safe tag, until safe tag block number is extracted and compared
     if (a == b) {
         return 0;
-    } else if (("genesis" == a || "earliest" == a || 0 == a) && ("genesis" == b || "earliest" ==  b || 0 == b)) {
+    } else if (("genesis" === a || "earliest" === a || 0 === a) && ("genesis" === b || "earliest" ===  b || 0 === b)) {
         return 0;
-    } else if ("genesis" == a || "earliest" == a) {
+    } else if ("genesis" === a || "earliest" === a || a === 0) {
         // b !== a, thus a < b
         return -1;
-    } else if ("genesis" == b || "earliest" == b) {
+    } else if ("genesis" === b || "earliest" === b || b === 0) {
         // b !== a, thus a > b
         return 1;
-    } else if (a == "latest") {
-        if (b == "pending") {
+    } else if (a === "latest" || a === "finalized") {
+        if (b === "pending") {
             return -1;
         } else {
             // b !== ("pending" OR "latest"), thus a > b
             return 1;
         }
-    } else if (b === "latest") {
-        if (a == "pending") {
+    } else if (b === "latest" || b === "finalized") {
+        if (a === "pending") {
             return 1;
         } else {
             // b !== ("pending" OR "latest"), thus a > b
             return -1
         }
-    } else if (a == "pending") {
+    } else if (a === "pending") {
         // b (== OR <) "latest", thus a > b
         return 1;
-    } else if (b == "pending") {
+    } else if (b === "pending") {
         return -1;
-    } else {
+    } else if(a === "safe") {
+        if( b === "finalized" || b === "latest" ||  b === "pending" ){
+            return -1;
+        }else {
+            return 1;
+        }
+    } else if(b === "safe") {
+        if( a === "finalized" || a === "latest" ||  a === "pending" ){
+            return -1;
+        }else {
+            return 1;
+        }
```

Comment:
Sorry, I think this condition has a typo. However, the code could never come to this section using our test cases because there are already redundant checks at pervious `if`s.  So, I think we need to rewrite those `if`s that comes before. Or refactor the last 2 `else if`. What do you think?

```suggestion
        if( a === "finalized" || a === "latest" ||  a === "pending" ){
            return 1;
        }else {
            return -1;
        }
```